File: app/src/main/java/com/abishov/hexocat/github/filters/SearchQuery.java
package com.abishov.hexocat.github.filters;

import static org.threeten.bp.format.DateTimeFormatter.ISO_LOCAL_DATE;

import androidx.annotation.Nullable;
import java.util.Objects;
import org.threeten.bp.LocalDate;

public final class SearchQuery {

  private final LocalDate createdSince;

  SearchQuery(Builder builder) {
    createdSince = Objects.requireNonNull(builder.createdSince);
  }

  @Override
  public String toString() {
    return "created:>=" + ISO_LOCAL_DATE.format(createdSince);
  }

  public static final class Builder {

    @Nullable
    private LocalDate createdSince;

    public Builder createdSince(LocalDate createdSince) {
      this.createdSince = createdSince;
      return this;
    }

    public SearchQuery build() {
      return new SearchQuery(this);
    }
  }
}

File: app/src/main/java/com/abishov/hexocat/github/filters/SearchQuery.kt
package com.abishov.hexocat.github.filters

import org.threeten.bp.LocalDate
import org.threeten.bp.format.DateTimeFormatter

class SearchQuery(
  private val createdSince: LocalDate,
  private val sort: Sort = Sort.STARS,
  private val order: Order = Order.DESC
) {

  override fun toString(): String {
    val dateString = DateTimeFormatter.ISO_LOCAL_DATE.format(createdSince)
    return "created:>=$dateString sort:$sort-$order"
  }
}
